### PR TITLE
add: xpack 1.0.2

### PR DESCRIPTION
**Pull request by Zvicii**
Specify library name and version: xpack / 1.0.2

xpack is a C++ reflection library, gives you the ability to convert between struct and json/xml. Its a header only library, so its very easy to use. I'm not the author, just thought its a good library to share.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Zvicii** [comment]
@SSE4 @uilianries @jgsogo @SpaceIm
please, review

**Comment by Zvicii** [comment]
Help needed! 
why does the ci not downloading source? And try to build a header-only lib？ I specify header_only() in package_id().
I try my best to copy rapidjson's recipe, I cant find the problem.

File: recipes/xpack/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    homepage = "https://github.com/xyz347/xpack"
+    license = "MIT"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+       tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (SSE4) [inline recipes/xpack/all/conanfile.py:21]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

File: recipes/xpack/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "include"))
```
Comment (SSE4) [inline recipes/xpack/all/conanfile.py:25]:
```suggestion
        self.copy(pattern="*", dst="include", src=self._source_subfolder))
```
seems like there is no include sub-directory, headers are within root

File: recipes/xpack/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+       tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
```
Comment (SSE4) [inline recipes/xpack/all/conanfile.py:24]: I don't see a license in tags 1.0.0 or 1.0.1, so it has to be downloaded separately, or extracted from sources

Comment (Zvicii) [reply recipes/xpack/all/conanfile.py:24]: yes I found what I was doing wrong, fixing it now.

File: recipes/xpack/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        self.copy(pattern="*", dst=os.path.join("include", "xpack"), src=self._source_subfolder)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "XPACK"
+        self.cpp_info.names["cmake_find_package_multi"] = "XPACK"
```
Comment (SpaceIm) [inline recipes/xpack/all/conanfile.py:32]:
```suggestion
```

there is no config file upstream

File: recipes/xpack/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst=os.path.join("include", "xpack"), src=self._source_subfolder)
```
Comment (SpaceIm) [inline recipes/xpack/all/conanfile.py:25]: That's too much. There are folders and files you don't want in include dir, like gtest, example, .gitignore etc

What about thirdparty folder? Is it possible to unvendor rapidjson & rapidxml?

File: recipes/xpack/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (SpaceIm) [inline recipes/xpack/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/xpack/all/conandata.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "cci.20220223":
```
Comment (SpaceIm) [inline recipes/xpack/all/conandata.yml:2]: Coul you ask to library author to make a new release? Last one is 1 year old: https://github.com/xyz347/xpack/releases

File: recipes/xpack/all/conanfile.py (hunk 5 above)
Comment (Zvicii) [reply recipes/xpack/all/conanfile.py]: I have exclude example and gtest dir. As for the thirdparty dir, I dont know if they are modified. Maybe I can exclude them and make them requirements.

File: recipes/xpack/all/conandata.yml (hunk 7 above)
Comment (Zvicii) [reply recipes/xpack/all/conandata.yml:2]: I can try.

File: recipes/xpack/all/conanfile.py (hunk 5 above)
Comment (Zvicii) [reply recipes/xpack/all/conanfile.py]: rapidjson and rapixml are included as "thirdparty/rapidjson/*" "thirdparty/rapidxml/*", so I cant get rid of them without making pr to xpack.

File: recipes/xpack/all/conandata.yml (hunk 7 above)
Comment (Zvicii) [reply recipes/xpack/all/conandata.yml]: I had the author release the version 1.0.2, I changed the conandata version to 1.0.2 as well

File: recipes/xpack/all/test_package/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/xpack/all/test_package/CMakeLists.txt:8]:
```suggestion
conan_basic_setup(TARGETS)

find_package(xpack CONFIG REQUIRED)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} xpack::xpack)
```

File: recipes/xpack/all/test_package/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/xpack/all/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_config"
```

**Review by uilianries — COMMENTED** [review]
Just improve test package a little bit

File: recipes/xpack/all/test_package/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_config"
```
Comment (jgsogo) [inline recipes/xpack/all/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

It was a typo from a previous review.

File: recipes/xpack/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+required_conan_version = ">=1.33.0"
+
+
+class XpackConan(ConanFile):
+    name = "xpack"
+    description = "C++ reflection, ability to convert between C++ structures and json/xml."
+    topics = ("xpack", "json", "reflection", "xml")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/xyz347/xpack"
+    license = "MIT"
```
Comment (uilianries) [inline recipes/xpack/all/conanfile.py:13]:
```suggestion
    license = "Apache-2.0"
```

The correct license is Apache 2.0 ! 

https://github.com/xyz347/xpack/blob/v1.0.2/LICENSE

**Review by uilianries — APPROVED** [review]

File: recipes/xpack/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst=os.path.join("include", "xpack"), excludes=["example/*", "gtest/*"], src=self._source_subfolder)
```
Comment (SpaceIm) [inline recipes/xpack/all/conanfile.py:25]: not sufficient, packaging is not very neat:

```
xpack/1.0.2 package(): Packaged 54 '.h' files
xpack/1.0.2 package(): Packaged 3 files: .gitignore, LICENSE, LICENSE
xpack/1.0.2 package(): Packaged 1 '.pri' file: xpack.pri
xpack/1.0.2 package(): Packaged 1 '.md' file: README.md
xpack/1.0.2 package(): Packaged 4 '.hpp' files: rapidxml_print.hpp, rapidxml_iterators.hpp, rapidxml.hpp, rapidxml_utils.hpp
xpack/1.0.2 package(): Packaged 1 '.html' file: manual.html
xpack/1.0.2 package(): Packaged 1 '.txt' file: license.txt
```

```suggestion
        self.copy(pattern="*.h", dst=os.path.join("include", "xpack"), excludes=["example/*", "gtest/*", "thirdparty/*"], src=self._source_subfolder)
        # TODO: remove bundled dependencies
        self.copy(pattern="*", dst=os.path.join("include", "xpack", "thirdparty"), src=os.path.join(self._source_subfolder, "thirdparty"))
```

**Review by uilianries — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**